src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/MulOperation.java

```diff
@@ -0,0 +1,95 @@
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of {@code $mul} update operation used to modify numeric field values in documents.
+ * See {@href https://www.mongodb.com/docs/manual/reference/operator/update/mul/} for full
+ * explanation.
+ */
+public class MulOperation extends UpdateOperation {
+  private final List<Action> actions;
+
+  private MulOperation(List<Action> actions) {
+    this.actions = sortByPath(actions);
+  }
+
+  public static MulOperation construct(ObjectNode args) {
+    Iterator<Map.Entry<String, JsonNode>> fieldIter = args.fields();
+
+    List<Action> updates = new ArrayList<>();
+    while (fieldIter.hasNext()) {
+      Map.Entry<String, JsonNode> entry = fieldIter.next();
+      // Verify we have neither operators...
+      String name = validateNonModifierPath(UpdateOperator.MUL, entry.getKey());
+      // nor try to change doc id
+      name = validateUpdatePath(UpdateOperator.MUL, name);
+      JsonNode value = entry.getValue();
+      if (!value.isNumber()) {
+        throw new JsonApiException(
+            ErrorCode.UNSUPPORTED_UPDATE_OPERATION_PARAM,
+            ErrorCode.UNSUPPORTED_UPDATE_OPERATION_PARAM.getMessage()
+                + ": $mul requires numeric parameter, got: "
+                + value.getNodeType());
+      }
+      updates.add(new Action(name, (NumericNode) value));
+    }
+    return new MulOperation(updates);
+  }
+
+  @Override
+  public boolean updateDocument(ObjectNode doc, UpdateTargetLocator targetLocator) {
+    boolean modified = false;
+    for (Action action : actions) {
+      final String path = action.path;
+      final NumericNode multiplier = action.value;
+
+      UpdateTarget target = targetLocator.findOrCreate(doc, path);
+      JsonNode oldValue = target.valueNode();
+
+      if (oldValue == null) { // No such property? Initialize as zero
+        target.replaceValue(doc.numberNode(BigDecimal.ZERO));
+        modified = true;
+      } else if (oldValue.isNumber()) { // Otherwise, if existing number, can modify
+        JsonNode newValue = multiply(doc, oldValue, multiplier);
+        if (!oldValue.equals(newValue)) {
+          target.replaceValue(newValue);
+          modified = true;
+        }
+      } else { // Non-number existing value? Fail
+        throw new JsonApiException(
+            ErrorCode.UNSUPPORTED_UPDATE_OPERATION_TARGET,
+            ErrorCode.UNSUPPORTED_UPDATE_OPERATION_TARGET.getMessage()
+                + ": $mul requires target to be Number; value at '"
+                + path
+                + "' of type "
+                + oldValue.getNodeType());
+      }
+    }
+
+    return modified;
+  }
+
+  private JsonNode multiply(ObjectNode doc, JsonNode oldValue, JsonNode multiplierValue) {
+    BigDecimal multiplier = multiplierValue.decimalValue();
+    BigDecimal old = oldValue.decimalValue();
+
+    // Short-cuts to avoid calculation
+    if (BigDecimal.ONE.equals(multiplier) || BigDecimal.ZERO.equals(old)) {
```

Comment:
Do we need this, cpu cost to multiplying by 1 or 0 vs running if condition every time.

Reply:
Probably not a huge difference either way, but doing check avoids document modification and keeps `JsonNode` comparison as identity comparison. But I guess it assumes that case of multiplier `1` or old value of `0` is common.

Actually, come to think of this, I think latter seems much more likely; so what I should do is remove first check, leave second.

Reply:
Changed to only check for old value being 0 (seems like common enough case), don't check on multiply by 1.